### PR TITLE
feat: Add support for PodDisruptionBudget

### PR DESCRIPTION
**Pull request by brasseld**
To have some context: https://innablr.com.au/blog/what-is-pod-disruption-budget-in-k8s-and-why-to-use-it/

### CODE REVIEW

File: cockpit/templates/api/api-pdb.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    app.kubernetes.io/component: "{{ .Values.api.name }}"
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+spec:
+  {{- if .Values.api.pdb.maxUnavailable }}
+  maxUnavailable: {{ .Values.api.pdb.maxUnavailable }}
+  {{ end -}}
+  {{- if .Values.api.pdb.minAvailable }}
+  minAvailable: {{ .Values.api.pdb.minAvailable }}
+  {{ end -}}
```
Comment (jgiovaresco) [inline cockpit/templates/api/api-pdb.yaml:19]: Shouldn't we drive the content with the Values?
Other we would need to upgrade the chart to support more settings 🤔 

```
 {{ toYaml .Values.api.pdb | nindent 4 -}}
```

File: cockpit/values.yaml [hunk 2]
```diff
@@ -193,6 +193,11 @@ api:
     periodSeconds: 30
     failureThreshold: 3
 
+  pdb:
+    enabled: false
+    minAvailable: ""
+    maxUnavailable: "50%"
```
Comment (jgiovaresco) [inline cockpit/values.yaml:199]: Do we need to set values if we disable pdb by default?

File: cockpit/templates/api/api-pdb.yaml (hunk 1 above)
Comment (brasseld) [reply cockpit/templates/api/api-pdb.yaml:19]: I would say why not, but PDB is very straightforward, not a lot you can do https://docs.openshift.com/container-platform/4.9/rest_api/policy_apis/poddisruptionbudget-policy-v1.html

File: cockpit/values.yaml (hunk 2 above)
Comment (brasseld) [reply cockpit/values.yaml:199]: Good question, I left it there for providing example...

Comment (jgiovaresco) [reply cockpit/values.yaml:199]: We could maybe put a comment with a link to the doc 🤷

**Review by jgiovaresco — APPROVED** [review]

**Pull request merged**